### DISCUSSION TITLE
TypeError: Request constructor: HEAD or GET Request cannot have a body

### DISCUSSION THREAD

**Discussion by petrparolek** [jiripudil/Naja#174] [Support] [answered]
Hello, I use Naja 2 first time and I migrating from nette.ajax.js. What I do wrong?
```
            data = {
                'year': 2020,
            };
            console.log(data);
            naja.makeRequest('GET', {link ChangeYear!}, data)
                .then(payload => {
                    console.log(payload);
                });
```
Error:
```
Uncaught (in promise) TypeError: Request constructor: HEAD or GET Request cannot have a body.
    makeRequest https://is.local/www/node_modules/naja/dist/Naja.js:2068
    loadData https://is.local/?do=paymentsJS-ChangeYear:396
    <anonymous> https://is.local/?do=paymentsJS-ChangeYear:402
    jQuery 13
```

**Comment by jiripudil** [comment, marked as answer]
Hello,

Naja sends the `data` as a request body which is not supported for `GET` and `HEAD` methods – for those, data should be sent as part of the URL in the query string.

I guess nette.ajax.js (or rather the underlying jQuery's $.ajax) puts given data into the query string for you? I think Naja could do the same, especially since it already does so when sending forms via `GET` 🤔 (edit: 👉 #175)

For the time being, I believe something like this should work:

```js
const url = new URL({link changeYear!});
url.searchParams.append('year', '2020');

naja.makeRequest('GET', url);
```

**Comment by petrparolek** [comment]
nette sends to latte relative URL not abolute `Uncaught TypeError: URL constructor: /?do=paymentsJS-changeYear is not a valid URL.`

**Reply by jiripudil** [reply]
Ah yes. The [URL's constructor](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) accepts an optional `base` argument:

```js
new URL({link changeYear!}, location.href);
```

**Comment by petrparolek** [comment]
thank you very much, it works
